### PR TITLE
Re-enable support for array-valued geo_shape fields.

### PR DESCRIPTION
**Pull request by talevy**
A regression in the mapping code led to geo_shape no longer supporting
array-valued fields. This commit fixes this support and adds an integration
test to make sure this problem does not return!

example:

```
PUT /example
{
    "mappings": {
        "properties": {
            "location": {
                "type": "geo_shape"
            }
        }
    }
}

PUT /example/_doc/1
{
    "location": [
        {
            "type": "LineString",
            "coordinates": [[100.0, 1.0], [101.0, 0.0]]
        },
        {
            "type": "LineString",
            "coordinates": [[100.0, 1.0], [101.0, 0.0]]
        }
    ]
}
```

original user issue posting: https://discuss.elastic.co/t/elastic-search-7-8-0-and-multiple-geoshapes/239350

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/search/geo/GeoShapeIntegrationIT.java [hunk 1]
```diff
@@ -136,6 +136,34 @@ public void testIgnoreMalformed() throws Exception {
         assertThat(searchResponse.getHits().getTotalHits().value, equalTo(1L));
     }
 
+    /**
+     * Test that geo_shape field can parse input as an array of values.
+     */
+    public void testArrayOfGeoShapeParsing() throws Exception {
```
Comment (jtibshirani) [inline server/src/internalClusterTest/java/org/elasticsearch/search/geo/GeoShapeIntegrationIT.java:142]: This test could also instead live in `GeoShapeFieldMapperTests`, which is closer to a unit test suite. It seems common to test document parsing logic in `*FieldMapperTests`.

File: server/src/main/java/org/elasticsearch/index/mapper/AbstractGeometryFieldMapper.java [hunk 2]
```diff
@@ -285,7 +285,7 @@ protected void parseCreateField(ParseContext context) throws IOException {
 
     @Override
     public final boolean parsesArrayValue() {
```
Comment (jtibshirani) [inline server/src/main/java/org/elasticsearch/index/mapper/AbstractGeometryFieldMapper.java:287]: I think we need to remove this override and instead  have `GeoPointFieldMapper` and `PointFieldMapper` implement it and return `true`?

**Review by jtibshirani — COMMENTED** [review]
Thanks so much @talevy for fixing this. I totally should have noticed it in #58786!

File: server/src/main/java/org/elasticsearch/index/mapper/AbstractGeometryFieldMapper.java (hunk 2 above)
Comment (talevy) [reply server/src/main/java/org/elasticsearch/index/mapper/AbstractGeometryFieldMapper.java:287]: woops. jumped on that too fast. you're right

File: server/src/internalClusterTest/java/org/elasticsearch/search/geo/GeoShapeIntegrationIT.java (hunk 1 above)
Comment (talevy) [reply server/src/internalClusterTest/java/org/elasticsearch/search/geo/GeoShapeIntegrationIT.java:142]: will do!

**Comment by talevy** [comment]
note: points and geo-points have unit tests for arrays!

**Review by iverase — APPROVED** [review]
lgtm. just wondering if we should add a test in `GeoShapeWithDocValuesFieldMapperTests` and in `LegacyGeoShapeFieldMapperTests`

**Pull request merged**